### PR TITLE
fix: consider child nodes while getting bin details

### LINKED ISSUE #28766: get_item_details() args.warehouse results in 0 actual_qty when group node is supplied
https://github.com/frappe/erpnext/blob/ef53eb7fcda8a58c62a11aef3c8a7dc16f5e3f0b/erpnext/stock/get_item_details.py#L29

If args.warehouse is a group node then https://github.com/frappe/erpnext/blob/ef53eb7fcda8a58c62a11aef3c8a7dc16f5e3f0b/erpnext/stock/get_item_details.py#L988 will return actual_qty 0

# Expected Result
return the sum of projected_qty, actual_qty, and reserved_qty  for all nodes under the specified group.

### PR DESCRIPTION
**Pull request by s-aga-r**
closes: https://github.com/frappe/erpnext/issues/28766

Before:

image

After:

image

image

### CODE REVIEW

File: erpnext/stock/get_item_details.py [hunk 1]
```diff
@@ -1171,14 +1175,28 @@ def get_projected_qty(item_code, warehouse):
...
+		bin_details = (
+			frappe.qb.from_(bin)
+			.select(
+				Sum(bin.projected_qty).as_("projected_qty"),
+				Sum(bin.actual_qty).as_("actual_qty"),
+				Sum(bin.reserved_qty).as_("reserved_qty"),
+			)
+			.where((bin.item_code == item_code) & (bin.warehouse.isin(warehouses)))
+		).run(as_dict=True)[0]
+		bin_details = {k: 0 if not v else v for k, v in bin_details.items()}
```
Comment (rohitwaghchaure) [inline erpnext/stock/get_item_details.py]:
```
bin_details = {k: 0 if not v else v for k, v in bin_details.items()}
```
What is the use of this? bin_details is already a dictionary

Comment (s-aga-r) [reply erpnext/stock/get_item_details.py]: This is to replace the `None` values with `0`. This happens when there is no Bin for the key (item_code, warehouse). 

Example: 

image

image

Comment (s-aga-r) [reply erpnext/stock/get_item_details.py]: Let's use `frappe.db.get_value()` instead.

Comment (rohitwaghchaure) [reply erpnext/stock/get_item_details.py]: Use COALESCE(SUM(projected_qty),0)

**Review by rohitwaghchaure — DISMISSED** [review]

**Pull request merged**